### ISSUE TITLE
Remove underscore prefix from (almost) all files

### ISSUE THREAD

**Issue by atilag** [Qiskit/qiskit-terra#1206]
### What is the expected enhancement?
As we are transitioning from relative imports to absolute imports:
```python
from ._module import Class ---> from qiskit.module import _module
```
We are introducing code like:
```python
circuit = _quantumcircuit.QuantumCircuit()
```
... this code is telling me that I want a `QuantumCircuit` class from an internal/private `_module`, but QuantumCircuit is not internal nor private.

The concept of internal classes has been diffused during all this time, and I don't think we have this concept anymore, at least not in the vast majority of the files.
So in order to keep code cleaner, I'd suggest to remove underscore prefix from all the files that we have doubts about its domain and files we know they are meant to be used by other components (I'd say all of them).

This is somewhat of a large change, mostly style related so I'm labeling this issue as `discussion` just in case someone has different views.

**Comment by charlesastaylor** [comment]
I could do this, once said discussion has happened of course!

**Comment by mtreinish** [comment]
I'm fine with doing this, but my only concern is the external messaging. Personally I don't see an issue with a leading underscore with an import inside the same repo, to me having a module with a leading underscore means we're not providing an external api stability contract on that module. If we're going to change that for all the modules marked as private we'll have to change how we develop things. Otherwise we'll have to make it very clear which interfaces we're defining as stable and which ones we're not.

**Comment by 1ucian0** [comment]
There are not that many currently:
```
$ grep -rI 'import _' qiskit | cut -f 2 -d ':' | sort | uniq
            from qiskit.tools.jupyter.jupyter_magics import _html_checker  # pylint
    from qiskit.tools.visualization._utils import _validate_input_state
from . import _node as node
from . import _schema_validation
from . import _util
from .version import __version__
from qiskit import __path__ as qiskit_path
from qiskit._util import _has_connection
from qiskit.circuit.decorators import _op_expand
from qiskit.qasm import _qasm
from qiskit.qasm._node import _node
from qiskit.tools.visualization import _latex
from qiskit.tools.visualization import _matplotlib
from qiskit.tools.visualization import _qcstyle
from qiskit.tools.visualization import _text
from qiskit.tools.visualization import _utils
from qiskit.tools.visualization._utils import _validate_input_state
from qiskit.version import __version__
```

**Comment by diego-plan9** [comment]
> Currently, there are not that many:

Actually that grepping seems to be missing some files - we are at ~70ish currently.

**Comment by 1ucian0** [comment]
What else is still pending here @diego-plan9 ?

**Comment by diego-plan9** [comment]
Should be finally closed after #2095 :tada:
